### PR TITLE
[20] Sign-Up Form Validation

### PR DESCRIPTION
**Pull request by opeyemipeter1759**
## Task: `Name of given task`
- `Changes you made to given task`
image
image
image

### CODE REVIEW

**Comment by opeyemipeter1759** [comment]
@ToluAkin @adaugochi kindly review

File: submissions/opeyemiPeter/DOM/script.js [hunk 1]
```diff
@@ -3,25 +3,24 @@
...
+function validateForm(e) {
+    if (email.value === "") {
+        document.querySelector(".err").innerHTML = "Please enter your Email Address";
+        e.preventDefault();
+        return false;
     }
-    submitForm.addEventListener('submit', validateForm);
\ No newline at end of file
+
+    if (!email.value.match(emailRegex)) {
```
Comment (adaugochi) [inline submissions/opeyemiPeter/DOM/script.js:16]: kindly use an `else if` here

File: submissions/opeyemiPeter/DOM/signup.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  confirmPassword = document.getElementById("confirm_password");
+
+function validation(field, regExp, errorMess1, errorMess2, event) {
+  event.preventDefault();
+  if (field.value === "") {
+    field.nextElementSibling.innerHTML = errorMess1;
+    event.preventDefault();
+    return false;
+  }
+  if (!field.value.match(regExp)) {
```
Comment (adaugochi) [inline submissions/opeyemiPeter/DOM/signup.js:16]: same here too

File: submissions/opeyemiPeter/DOM/signup.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  } else {
+    field.nextElementSibling.innerHTML = "";
+    return true;
+  }
+}
+
+function passwordValidation(password1, password2, event) {
+  const passwordRegex = /(?=.*\d)(?=.*[a-z])(?=.*[a-zA-Z]).{6,}$/;
+
+  if ((password1.value && password2.value) === "") {
```
Comment (adaugochi) [inline submissions/opeyemiPeter/DOM/signup.js:29]: I tot I spoke about this in the chat we had. This condition can be handled in the function above

File: submissions/opeyemiPeter/DOM/signup.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    password1.nextElementSibling.innerHTML = "Please enter your password";
+    password2.nextElementSibling.innerHTML = "Please confirm your password";
+    event.preventDefault();
+    return false;
+  }
+  if (
+    !(password1.value.match(passwordRegex)) &&
```
Comment (adaugochi) [inline submissions/opeyemiPeter/DOM/signup.js:36]: This condition can be handled in the function above

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/DOM/signup.js (hunk 3 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/DOM/signup.js:29]: yes, you did, but it didnt work out. that why i resorted into this. due to the number parameters the password function expects to passed in

File: submissions/opeyemiPeter/DOM/script.js (hunk 1 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/DOM/script.js:16]: alright, right away

File: submissions/opeyemiPeter/DOM/signup.js (hunk 3 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/DOM/signup.js:29]: @adaugochi , i will appreciate if i can see how to implement it, the password validation function takes in 3 parameters while the validation function takes in 5 parameters so i dont know how to rightly implement, but i dont understand the concept just the implementation.

i have made the  'else if 'correction.
thanks

Comment (adaugochi) [reply submissions/opeyemiPeter/DOM/signup.js:29]: It also takes 5. `validation(password1, passwordRegex, 'Please enter your password', 'Your password should have a minimum of 6 characters, 1 capital letter, 1 special character eg @ and 1 number', event)`. How is it not taking 5?

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/DOM/signup.js]: @adaugochi thanks i changed it now, you can now view.

File: submissions/opeyemiPeter/DOM/signup.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  );
+  validation(
+    formEmail,
+    emailRegex,
+    "Please enter your Email",
+    "Please enter a valid Email",
+    event
+  );
+  validation(
+    password,
```
Comment (adaugochi) [inline submissions/opeyemiPeter/DOM/signup.js:77]: You didn't do for confirm password

File: submissions/opeyemiPeter/DOM/script.js (hunk 1 above)
Comment (adaugochi) [reply submissions/opeyemiPeter/DOM/script.js:16]: You haven't fixed this either

File: submissions/opeyemiPeter/DOM/signup.js (hunk 5 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/DOM/signup.js:63]: done.

File: submissions/opeyemiPeter/DOM/script.js (hunk 1 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/DOM/script.js:16]: @adaugochi i have done that now.

File: submissions/opeyemiPeter/DOM/signup.js (hunk 5 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/DOM/signup.js:63]: @adaugochi  kindly checked now.

**Comment by opeyemipeter1759** [comment]
@adaugochi kindly review now

**Review by adaugochi — APPROVED** [review]

**Pull request merged**